builtin/providers/cloudflare/resource_cloudflare_record_migrate_test.go

```diff
@@ -0,0 +1,291 @@
+package cloudflare
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"testing"
+
+	"github.com/cloudflare/cloudflare-go"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestCloudFlareRecordMigrateState(t *testing.T) {
+	// create the test server for mocking the API calls
+	ts := mockCloudFlareEnv()
+	defer ts.Close()
+
+	// Create a CloudFlare client, overriding the BaseURL
+	cfMeta, err := cloudflare.New(
+		"sometoken",
+		"someemail",
+		mockHTTPClient(ts.URL),
+	)
+
+	if err != nil {
+		t.Fatalf("Error building CloudFlare API: %s", err)
+	}
+
+	cases := map[string]struct {
+		StateVersion int
+		ID           string
+		Attributes   map[string]string
+		Expected     string
+		ShouldFail   bool
+	}{
+		"ttl_120": {
+			StateVersion: 0,
+			ID:           "123456",
+			Attributes: map[string]string{
+				"id":       "123456",
+				"name":     "notthesub",
+				"hostname": "notthesub.hashicorptest.com",
+				"type":     "A",
+				"content":  "10.0.2.5",
+				"ttl":      "120",
+				"zone_id":  "1234567890",
+				"domain":   "hashicorptest.com",
+			},
+			Expected: "7778f8766e583af8de0abfcd76c5dAAA",
+		},
+		"ttl_121": {
+			StateVersion: 0,
+			ID:           "123456",
+			Attributes: map[string]string{
+				"id":       "123456",
+				"name":     "notthesub",
+				"hostname": "notthesub.hashicorptest.com",
+				"type":     "A",
+				"content":  "10.0.2.5",
+				"ttl":      "121",
+				"zone_id":  "1234567890",
+				"domain":   "hashicorptest.com",
+			},
+			Expected: "5558f8766e583af8de0abfcd76c5dBBB",
+		},
+		"mx_priority": {
+			StateVersion: 0,
+			ID:           "123456",
+			Attributes: map[string]string{
+				"id":       "123456",
+				"name":     "hashicorptest.com",
+				"type":     "MX",
+				"content":  "some.registrar-servers.com",
+				"ttl":      "1",
+				"priority": "20",
+				"zone_id":  "1234567890",
+				"domain":   "hashicorptest.com",
+			},
+			Expected: "12342092cbc4c391be33ce548713bba3",
+		},
+		"mx_priority_mismatch": {
+			StateVersion: 0,
+			ID:           "123456",
+			Attributes: map[string]string{
+				"id":       "123456",
+				"type":     "MX",
+				"name":     "hashicorptest.com",
+				"content":  "some.registrar-servers.com",
+				"ttl":      "1",
+				"priority": "10",
+				"zone_id":  "1234567890",
+				"domain":   "hashicorptest.com",
+			},
+			Expected:   "12342092cbc4c391be33ce548713bba3",
+			ShouldFail: true,
+		},
```

Comment:
Nice!

Looks like these exercise TTL and Priority matching - need another pair for exercising the "proxied" logic too?